src/main/java/core/basesyntax/PurchaseOperation.java

```diff
@@ -0,0 +1,8 @@
+package core.basesyntax;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void handle(ShopStorage storage, FruitTransaction transaction) {
+        storage.updateFruitQuantity(transaction.getFruit(), -transaction.getQuantity());
```

Comment:
Before updating the fruit quantity, check if the resulting balance would be negative. If it is, throw a `RuntimeException` to prevent the storage from having a negative quantity. This aligns with the checklist requirement: 'Check result balance before saving it in the Storage - it should be positive. Throw `RuntimeException` in case the balance is negative.'